packages/core/src/interpreter.ts

```diff
@@ -587,15 +587,21 @@ export class Interpreter<
         child.stop();
       }
     });
+    this.children.clear();
 
     // Cancel all delayed events
     for (const key of Object.keys(this.delayedEventsMap)) {
       this.clock.clearTimeout(this.delayedEventsMap[key]);
     }
 
     this.scheduler.clear();
+    this.scheduler = new Scheduler({
```

Comment:
this assigns a fresh scheduler because the old one already has `this.initialized === true` and thus when we start to send events to a stopped machine the processing in the scheduler goes wrong (it process the event instead of queeing it). This happens because the "active" states of the interpreter and the scheduler are not "synced"~

Reply:
Should this instead be assigned in `this.start(...)`? Reason being that for use-cases outside of React, a stopped service should be stopped, and not allocate a new scheduler at all, since it should no longer receive events. Maybe a micro-optimization, but we should allocate this object in `.start()` instead of `.stop()`.

Reply:
We can't do that because we need to start buffering events sent to a "stopped" service (that we treat as not started) before the `.start()` is even called.

So we either create this here or we hack this stuff around and assign this private property from within `@xstate/react`'s code. In `@xstate/react` we could add it here:
https://github.com/statelyai/xstate/blob/360e8546298c4a06b6d51d8f12c0563672dd7acf/packages/xstate-react/src/useInterpret.ts#L143-L146